### PR TITLE
Disallow sharing children of shared pools

### PR DESCRIPTION
**Pull request by vritant**

### CODE REVIEW

**Comment by vritant** [comment]
@candlepin-pull-request-bot test this please

**Comment by vritant** [comment]
@candlepin-pull-request-bot test this please

File: server/src/main/java/org/candlepin/policy/js/entitlement/EntitlementRulesTranslator.java [hunk 1]
```diff
@@ -188,7 +188,8 @@
         POOL_ERROR_MESSAGES.put(PoolErrorKeys.VIRTUAL_GUEST_RESTRICTED,
             I18n.marktr("Pool is restricted to virtual guests in their first day of existence: ''{1}''"));
         POOL_ERROR_MESSAGES.put(PoolErrorKeys.SHARING_A_SHARE,
-            I18n.marktr("Pool ''{1}'' is a shared pool and sharing a shared pool is prohibited."));
+            I18n.marktr("Pool ''{1}'' is a pool created because of a share and sharing a shared pool is " +
+            "prohibited."));
```
Comment (wottop) [inline server/src/main/java/org/candlepin/policy/js/entitlement/EntitlementRulesTranslator.java]: 2 sentences, just for the sake of readability.

File: server/src/main/java/org/candlepin/model/Pool.java [hunk 2]
```diff
@@ -117,8 +117,12 @@
         /** Attribute used to determine which specific host the pool was created for */
         public static final String REQUIRES_HOST = "requires_host";
 
-        /** Attribute used to determine if a pool was created by sharing */
-        public static final String SHARE = "share_derived";
+        /** Attribute used to determine if a pool was created by a share */
+        public static final String SHARED_POOL = "shared_pool";
```
Comment (wottop) [inline server/src/main/java/org/candlepin/model/Pool.java]: Is this comment accurate? You have a CREATED_BY_SHARE. How is SHARED_POOL different?.

File: server/src/main/java/org/candlepin/policy/js/entitlement/EntitlementRulesTranslator.java (hunk 1 above)
Comment (vritant) [inline server/src/main/java/org/candlepin/policy/js/entitlement/EntitlementRulesTranslator.java]: @wottop fixed

File: server/src/main/java/org/candlepin/model/Pool.java (hunk 2 above)
Comment (vritant) [inline server/src/main/java/org/candlepin/model/Pool.java]: @wottop hope my update clarifies the difference.

File: server/src/main/java/org/candlepin/model/Pool.java [hunk 3]
```diff
@@ -1262,8 +1273,8 @@ public PoolType getType() {
             if (hasAttribute(Attributes.UNMAPPED_GUESTS_ONLY)) {
                 return PoolType.UNMAPPED_GUEST;
             }
-            else if (hasAttribute(Attributes.SHARE)) {
-                return PoolType.SHARE_DERIVED;
+            else if (hasAttribute(Attributes.SHARED_POOL)) {
```
Comment (wottop) [inline server/src/main/java/org/candlepin/model/Pool.java]: hasAttribute and value = 'true'?
I realize you just substituted the tag, but the hasAttribute test seems insufficient. Same for unmapped guest condition.

**Comment by mstead** [comment]
@vritant Please add more context to you commit message. For example, why are we doing this?

**Comment by vritant** [comment]
replaced by #1693

**Pull request closed without merging**